server/service/oauth.go

```diff
@@ -88,12 +98,83 @@ func (o *OAuth) GetUser(ctx context.Context, code string) (*model.User, error) {
 	var user githubUser
 	err = json.NewDecoder(resp.Body).Decode(&user)
 	if err != nil {
-		return nil, fmt.Errorf("can't parse github response: %s", err)
+		return nil, fmt.Errorf("can't parse github user response: %s", err)
+	}
+
+	if o.restrictAccess != "" {
+		if err := o.checkAccess(client, o.restrictAccess, user.Login); err != nil {
+			return nil, err
+		}
 	}
+
+	role := model.Requester
+	if o.restrictReviewerAccess != "" {
+		err := o.checkAccess(client, o.restrictReviewerAccess, user.Login)
+		if err != nil && err != NoAccessError {
+			return nil, err
+		}
+		if err == NoAccessError {
+			role = model.Worker
+		}
+	}
+
 	return &model.User{
 		ID:        user.ID,
 		Login:     user.Login,
 		Username:  user.Username,
 		AvatarURL: user.AvatarURL,
+		Role:      role,
 	}, nil
 }
+
+const orgPrefix = "org:"
+const teamPrefix = "team:"
+
+func (o *OAuth) checkAccess(client *http.Client, restriction, login string) error {
+	if strings.HasPrefix(restriction, orgPrefix) {
+		org := strings.TrimPrefix(restriction, orgPrefix)
+		url := fmt.Sprintf("https://api.github.com/orgs/%s/members/%s", org, login)
+		resp, err := client.Get(url)
+		if err != nil {
+			return fmt.Errorf("can't get user organizations from github: %s", err)
+		}
+		// StatusNoContent means user is a member
+		// https://developer.github.com/v3/orgs/members/#check-membership
+		if resp.StatusCode != http.StatusNoContent {
+			return NoAccessError
+		}
+		return nil
+	}
+	if strings.HasPrefix(restriction, teamPrefix) {
```

Comment:
we (at src-d) are separating code blocks with a new line in between.
([examples in gogit](https://github.com/src-d/go-git/blob/master/remote.go))

Reply:
if we do that, we need linter for it. And please describe which blocks should be separated. From reading the source code of go-git I don't always see the pattern.

Reply:
Most of our code is always written that way (with NewLine after every code block).
(random files, also in borges... https://github.com/src-d/borges/blob/master/archiver.go https://github.com/src-d/borges/blob/master/common.go)

(we would also need a linter to ensure the opposite; the reason is it's an internal convention we decided to follow...)

The rule could be:
>_a NewLine after `}` if it's not followed by another `}`_